### DISCUSSION TITLE
App shows blank screen on gh-pages when using the homepage in package.json

### DISCUSSION THREAD

**Discussion by ayushi186** [community/community#148211] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

Hey Community ,
I am deploying my react app into gh-pages for hosting and as suggested by many posts I have tried the following already -
1. I am using react router and using createHashrouter in place of createbrowserrouter - index.tsx of project.
2. gave the basename as the <repo> name - please check the index.tsx of the project .
3. the homepage is - "homepage": "https://ayushi186.github.io/FrontEndQuizApp",
4. private has been set to false in package.json.

Even after following these steps I am getting blank screen .
Seems like my build index.html is not pointing to the proper file location . But I am not sure how this works .
My project is here - https://github.com/ayushi186/FrontEndQuizApp 
project after deployment - https://ayushi186.github.io/FrontEndQuizApp/
any pointers are highly appreciated .

Kind Regards,
Ayushi

**Comment by mecodeatlas** [comment]
Hey @ayushi186, thanks for reaching out!

I accessed the link you sent, and the website loaded normally without any blank pages. Are you still experiencing the same issue? If you've already resolved it, could you share how you fixed it?

Here's an image of how it works:

![Screenshot 2025-01-03 at 19 24 35](https://github.com/user-attachments/assets/58ba6d6d-59dc-4e28-8ab8-14180e67883d)

**Reply by ayushi186** [reply, marked as answer]
Hey codeatlas ,
yep the issue is now resolved and the resolution is as follows -
with the hashrouter we don't need basename property(in my case) which I was providing earlier so, it was assuming the relative path based on basename of creatHashRouter .

**Comment by server-mm404** [comment]
> ### Select Topic Area
> Question
> 
> ### Body
> Hey Community , I am deploying my react app into gh-pages for hosting and as suggested by many posts I have tried the following already -
> 
> 1. I am using react router and using createHashrouter in place of createbrowserrouter - index.tsx of project.
> 2. gave the basename as the  name - please check the index.tsx of the project .
> 3. the homepage is - "homepage": "https://ayushi186.github.io/FrontEndQuizApp",
> 4. private has been set to false in package.json.
> 
> Even after following these steps I am getting blank screen . Seems like my build index.html is not pointing to the proper file location . But I am not sure how this works . My project is here - https://github.com/ayushi186/FrontEndQuizApp project after deployment - https://ayushi186.github.io/FrontEndQuizApp/ any pointers are highly appreciated .
> 
> Kind Regards, Ayushi
